scripts/agent-check.sh

```
@@ -0,0 +1,101 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+echo "🔍 Running agent-friendly Rust checks..."
+
+TEMP_DIR=$(mktemp -d)
+trap "rm -rf $TEMP_DIR" EXIT
```

Shellcheck (0.10.0)

[warning] Use single quotes, otherwise this expands now rather than when signalled.

(SC2064)